Add ScaleInstrumentComponent Algorithm (#37909)
mantidproject/mantid: docs/source/algorithms/ScaleInstrumentComponent-v1.rst

.. algorithm::

.. summary::

.. relatedalgorithms::

.. properties::

Description
-----------

> This scales the detecors and pixels in a component. This can effectively expand or contract the overall dimensions

Reviewer: ```suggestion
This algorithm scales the detector pixels in a component. This can effectively expand or contract the overall dimensions
```

of the component by resizing its detector pixels, without moving its geometrical center. You can choose to scale the component

> without scaling the size of the detector pixels. This would allow you to expand or contract the detectors without changing the

Reviewer: ```suggestion
without scaling the size of the detector pixels. This would allow you to expand or contract the detector positions without changing the
```

size of the pixels.

The vector position of each detector is determined by :math:`Mr + (1 - M)R` where :math:`M` is a diagonal matrix with diagonal components given by parameter ``scalings``,

> :math:`R` is the position of the component, and :math:`r` is the position of the detector.

Reviewer: ```suggestion
:math:`R` is the geometrical center of the component, and :math:`r` is the position of the detector.
```
